pallets/pooled-staking/src/lib.rs

```diff
@@ -754,4 +757,40 @@ pub mod pallet {
             pools::distribute_rewards::<T>(&candidate, rewards)
         }
     }
+    impl<T: Config> tp_traits::StakingCandidateHelper<Candidate<T>> for Pallet<T> {
+        fn is_candidate_selected(candidate: &Candidate<T>) -> bool {
+            <SortedEligibleCandidates<T>>::get()
+                .into_iter()
+                .any(|c| &c.candidate == candidate)
+        }
+        fn on_online_status_change(
+            candidate: &Candidate<T>,
+            _is_online: bool,
+        ) -> DispatchResultWithPostInfo {
+            Calls::<T>::update_candidate_position(&[candidate.clone()])
+        }
+
+        #[cfg(feature = "runtime-benchmarks")]
+        fn make_collator_eligible_candidate(collator: &Candidate<T>) {
+            use sp_std::vec;
+            let minimum_stake = T::MinimumSelfDelegation::get();
+            T::Currency::set_balance(collator, minimum_stake + minimum_stake);
+            T::EligibleCandidatesFilter::make_candidate_eligible(collator, true);
+            Calls::<T>::request_delegate(
```

Comment:
Make sure fallible calls succeed using `.expect()`, otherwise if it breaks for some reason it'll be harder to find the cause.

Reply:
Good point! Updated them in https://github.com/moondance-labs/tanssi/pull/881/commits/3a116ecd6275c2d5b05cb220118b2cf019298151